apps/src/templates/TimeAgo.jsx

```diff
@@ -13,6 +13,38 @@ class TimeAgo extends React.Component {
     style: PropTypes.object
   };
 
+  state = {
+    timeout: undefined
+  };
+
+  componentDidMount() {
+    this.startTimer();
+  }
+
+  componentWillUnmount() {
+    this.clearTimer();
+  }
+
+  startTimer = () => {
+    // Update the display string once a minute.
+    //
+    // Note this component could be much more clever; if the given date is
+    // within a few minutes of now, update this quickly, but if it's a couple
+    // hours ago update much less frequently and if it's even longer ago then
+    // maybe don't bother updating at all.
+    //
+    // For now, that seems like a unnecessary optimization.
+    const timeout = setTimeout(this.startTimer, 60 * 1000);
+    this.setState({timeout});
+  };
+
+  clearTimer = () => {
+    clearTimeout(this.state.timeout);
+    this.setState({
+      timeout: undefined
+    });
```

Comment:
Why set the state when the component is about to unmount?

Reply:
In the abstract, we should clear the reference to the timeout when we clear the timeout itself. This implementation only clears the timeout in a scenario where the reference will also incidentally be cleared, but this makes it explicit. And I don't think (please correct me if I'm wrong!) that unnecessary `setState`s are actually harmful.

Reply:
Good point — `clearTimer` is only called from `componentWillUnmount` today, but that's not always guaranteed to be the case! As a super tiny style nit, I'd maybe do this for simplicity:

```js
componentWillUnmount() {
  clearTimeout(this.state.timeout);
}
```